File: Bit Manipulation/1442. Count Triplets That Can Form Two Arrays of Equal XOR/1442.py
class Solution:
    def countTriplets(self, arr: List[int]) -> int:
        res = 0
        
        for i in range(len(arr) - 1):
            a = 0
            for j in range(i + 1, len(arr)):
                a ^= arr[j-1]
                b = 0
                for k in range(j, len(arr)):
                    b ^= arr[k]
                    if a == b:
                        res += 1

        return res

File: Bit Manipulation/1442. Count Triplets That Can Form Two Arrays of Equal XOR/1442.cpp
#pragma GCC optimize("O3","Oz","unroll-loops","inline-functions","fast-math","march=native","lto","omit-frame-pointer")
class Solution {
public:
    int countTriplets(vector<int>& arr) {
        ios_base::sync_with_stdio(0);
        cin.tie(0);
        cout.tie(0);

        int res = 0;
        for (int i=0; i<arr.size()-1; i++) {
            int a = 0;
            for (int j=i+1; j<arr.size(); j++) {
                a^=arr[j-1];
                int b=0;
                for (int k=j; k<arr.size(); k++) {
                    b^=arr[k];
                    if (a==b) res++;
                }
            }
        }

        return res;
    }
};